apps/s_client.c

```diff
@@ -2218,7 +2219,20 @@ int s_client_main(int argc, char **argv)
                            "Didn't find STARTTLS in server response,"
                            " trying anyway...\n");
             BIO_printf(sbio, "STARTTLS\r\n");
-            BIO_read(sbio, sbuf, BUFSIZZ);
+            mbuf_len = BIO_read(sbio, mbuf, BUFSIZZ);
+            if (mbuf_len < 0) {
+                BIO_printf(bio_err, "BIO_read failed\n");
+                goto end;
+            }
+            mbuf[mbuf_len] = '\0';
+            if (sscanf(mbuf, "%3d ", &numeric) != 1) {
+                BIO_printf(bio_err, "Server does not support STARTTLS.\n");
+                goto shut;
+            }
+            if (numeric != 382) {
```

Comment:
Specification is that specific about number, huh? But real question why not just `strcmp(mbuf,"382")`? Or should it be "382 "? Thing here is that it would make sense to print what server returned, even in above case, i.e. when sscanf returns 0. I mean as it is now reply that doesn't start with number would be hidden from user. And since this code is effectively for *trouble-shooting* purposes, it should print everything server returns, even [or especially?] if it's not compliant with specification. And single strcmp would cover all that :-)